MoTrPAC/motrpac-frontend
# MOTR-269 Release Sample Count Visualization

### Key Changes

1. Replaced mocked upload/download stat UIs with UIs representing actual release sample counts in **Dashboard** component.
2. Updated **Sidebar** component with functionality to expand/collapse sidebar.
3. Added the rendering of tooltips for active sidebar items when the sidebar is collapsed.
4. Updated all pages that required authentication to support the collapsible sidebar.
5. Updated unit test assertions due to changes of various UI components.
6. Upgraded some `npm` dependencies.

### Steps to test sidebar

1. Sign into local instance of this branch.
2. Expect to see sidebar to be collapsed by default.
3. Click on the _hamburger_ menu icon at the top of the sidebar. Expect to see the sidebar being expanded.
4. Then click on the _close_ icon at the top of the sidebar. Expect to see the sidebar being collapsed.
5. You may wish to repeat the steps above on other pages.

### Steps to test dashboard
1. Sign into local instance of this branch and go to the `^/dashboard` page.
2. Expect to see the **Internal Release** and **PASS1A 6-Month** button being active by default at the upper-right corner.
3. Click on the **PASS1B 6-Month** button and expect to see the UIs (e.g. various metrics, sample count plot/table) being updated on the dashboard.
4. Click on the **Samples** dropdown menu above the plot and select **Tissue by count**. Expect to see the plot being re-rendered.
5. Click on the **Sort** dropdown menu above the table and select **Tissue name ascending**. Expect to see the table being sorted by tissue name in alphabetical order.
